### PR TITLE
create a subscription method

### PR DESCRIPTION
**Pull request by andymckay**
Requires https://github.com/mozilla/solitude/pull/392. Fixes mozilla/payments#18.

### CODE REVIEW

**Comment by andymckay** [comment]
missing a directory :(

File: docs/topics/braintree.rst [hunk 1]
```diff
@@ -187,3 +181,86 @@ Information about the payment method is stored in solitude.
...
+                "modified": "2015-05-06T13:34:53.763",
+                "paymethod": "/braintree/mozilla/paymethod/1/",
+                "provider_id": "some:id",
+                "resource_pk": 1,
+                "resource_uri": "/braintree/mozilla/subscription/1/",
+                "seller_product": "/generic/product/1/"
+            }
+        }
+
+    :>json string braintree id: id of the payment method in braintree.
```
Comment (kumar303) [inline docs/topics/braintree.rst]: id of the subscription, not payment method

File: docs/topics/braintree.rst [hunk 2]
```diff
@@ -187,3 +181,86 @@ Information about the payment method is stored in solitude.
...
+------------
+
+Creates a subscription in Braintree and the corresponding subscription in
+solitude.
+
+.. http:post:: /braintree/subscription/
+
+    :<json string paymethod: the uri of the payment method.
+    :<json string plan: the id of the plan being purchased.
+
```
Comment (kumar303) [inline docs/topics/braintree.rst]: In other docs I liked how you linked this ``mozilla`` output to the section that explains it in detail. In this case I think it should link to the ``/braintree/mozilla/subscription/<id>`` doc section.

File: docs/topics/braintree.rst [hunk 3]
```diff
@@ -102,6 +97,8 @@ object and Braintree customer.
     :query buyer: the primary key of the buyer.
     :query active: the active status.
 
+.. _payment-methods-label:
```
Comment (kumar303) [inline docs/topics/braintree.rst]: adding ``-label`` is redundant, no?

File: docs/topics/braintree.rst [hunk 4]
```diff
@@ -187,3 +181,86 @@ Information about the payment method is stored in solitude.
...
+        }
+
+    :>json boolean active: active flag for the method.
+    :>json string provider_id: an id for the payment method on the provider,
+                               this field is read only.
+    :>json string paymethod: the URI to `a payment object <payment-methods-label>`_.
+    :>json string seller_product: the URI to `a seller product
+                                  <seller-product>`_.
+
+.. http:patch:: /generic/subscription/<subscription id>/
```
Comment (kumar303) [inline docs/topics/braintree.rst]: Shouldn't this be ``/braintree/mozilla/subscription``?

File: docs/topics/braintree.rst [hunk 5]
```diff
@@ -187,3 +181,86 @@ Information about the payment method is stored in solitude.
...
+
+    :<json boolean active: if the buyer is currently active.
```
Comment (kumar303) [inline docs/topics/braintree.rst]: I think you mean subscription not buyer?

File: lib/buyers/models.py [hunk 6]
```diff
@@ -62,3 +64,29 @@ def incr_lockout(self):
...
+        (PAYMENT_METHOD_OPERATOR, PAYMENT_METHOD_OPERATOR),
+    ))
+    # Details about the type, eg Amex, Orange.
+    type_name = models.CharField(max_length=255)
+    # For credit cards, this is the last 4 numbers, could be a truncated
+    # phone number or paypal account for example.
+    truncated_id = models.CharField(max_length=255)
+
+    class Meta(Model.Meta):
+        db_table = 'buyer_payment_method'
```
Comment (kumar303) [inline lib/buyers/models.py]: where is ``buyer_payment_method`` defined? All I see is a ``braintree_pay_method`` table

Comment (andymckay) [inline lib/buyers/models.py]: i'm confused why this is even showing up in this PR, fixing

File: docs/topics/braintree.rst (hunk 3 above)
Comment (andymckay) [inline docs/topics/braintree.rst]: it is, but i've found being able to grep label and find all the labels for cross referencing really useful

**Comment by andymckay** [comment]
Sorry that merge went horribly wrong and I didn't help myself by not deleting the old stuff from lib/buyers when I moved it over. This one looks a bit happier.

**Comment by kumar303** [comment]
looks great! r+wc

**Pull request merged**